src/React/Resources/assets/src/render_controller.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import React, {ReactElement} from 'react';
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    readonly componentValue: string;
+    readonly propsValue: object;
+
+    static values = {
+        component: String,
+        props: Object,
+    };
+
+    connect() {
+        this._dispatchEvent('react:connect', { component: this.componentValue, props: this.propsValue });
+
+        // Use a timeout to avoid mounting and demounting right after
+        if ((this.element as any).timeout) {
+            clearTimeout((this.element as any).timeout);
+        }
+
+        (this.element as any).timeout = setTimeout(() => {
+            const component = window.resolveReactComponent(this.componentValue);
+            this._renderReactElement(React.createElement(component, this.propsValue, null));
+
+            this._dispatchEvent('react:mount', { component: this.componentValue, props: this.propsValue });
+        }, 50);
+    }
+
+    disconnect() {
+        (this.element as any).unmount();
+        this._dispatchEvent('react:unmount', { component: this.componentValue, props: this.propsValue });
+    }
+
+    _renderReactElement(reactElement: ReactElement) {
+        if (parseInt(React.version) >= 18) {
+            // React 18+ rendering
+            const root = require('react-dom/client').createRoot(this.element);
+            root.render(reactElement);
+
+            // Register a way to unmount this element
+            (this.element as any).unmount = () => {
+                root.unmount();
+            };
+
+            return;
+        }
+
+        // Up to React 17 rendering
+        const reactDom = require('react-dom');
+        reactDom.render(reactElement, this.element);
+
+        // Register a way to unmount this element
+        (this.element as any).unmount = () => {
+            reactDom.unmountComponentAtNode(this.element);
+        };
+    }
+
+    _dispatchEvent(name: string, payload: any) {
+        this.element.dispatchEvent(new CustomEvent(name, { detail: payload }));
```

Comment:
Make this bubble: https://github.com/symfony/ux/blob/040e75d55b4c8c267ffef1b61cc2f8670974ab96/src/Typed/Resources/assets/src/controller.ts#L65